src/platform/qt/Window.cpp

```diff
@@ -580,6 +580,15 @@ std::function<void()> Window::openTView(A... arg) {
 	};
 }
 
+template <typename T, typename... A>
+std::function<void()> Window::openTViewModal(A... arg) {
```

Comment:
Instead of having yet more similarly named mostly-overlapping functions we might want to add a keepalive flag to `openTView` if the only difference is the WA_DeleteOnClose, just like I did to the other functions. Wonder if there's some nicer way to handle this.

Reply:
That isn't the only difference. That one's just incidental to make sure the memory gets cleaned up. The core change is the use of `QDialog::open()` to make the popup modal, This solves the problem of having multiple windows open by blocking the user from getting to the command that opened the popup in the first place. I only chose this solution for the popups that can be treated as standalone behaviors where you're going in to do a specific thing and then you're done. I didn't do it for anything where being modal wouldn't make sense. (This is also the answer to the other question you asked in this function.)